src/_pytest/config/__init__.py

```diff
@@ -1192,17 +1201,33 @@ def _getini(self, name: str) -> Any:
         except KeyError:
             raise ValueError("unknown configuration value: {!r}".format(name))
         override_value = self._get_override_ini_value(name)
-        if override_value is None:
-            try:
-                value = self.inicfg[name]
-            except KeyError:
+        append_values = self._get_append_ini_values(name, type)
```

Comment:
Perhaps we can move this logic here to a function:


```python
def resolve_ini_option(ini_options: Dict, name: str, override_value, append_values) -> object:
    ...  
```

Which would then be called at this point like this:

```python
value = resolve_ini_option(self.inicfg, name, override_value, append_values)
```

And unittest that? I'm afraid the logic is getting convoluted enough that we should have proper tests for this functionality specifically.

Reply:
I agrree, let me refactor this function as part of this PR